### ISSUE TITLE
Mount pod keeps restarting

### ISSUE THREAD

**Issue by Haivilo** [juicedata/juicefs#5060]
<!--
Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
-->

**What happened**:
Mount pod keeps restarting:
```
juicefs-gke-bc-bc-staging-bc-bc-staging-gpu-l-e49df90a-66f6-pvc-fe0548b2-6a6f-4729-8a3e-155018ec5777-tsainc   1/1     Running   383 (3m38s ago)   16d
```

Logs don't contain useful information:
```
➜  ~ k logs juicefs-gke-bc-bc-staging-bc-bc-staging-gpu-l-e49df90a-66f6-pvc-fe0548b2-6a6f-4729-8a3e-155018ec5777-tsainc -n juicefs -p        
Defaulted container "jfs-mount" out of: jfs-mount, jfs-format (init)
2024/08/05 20:18:47.065203 juicefs[7] <INFO>: Meta address: redis://:****@...:6379/1 [interface.go:497]
2024/08/05 20:18:47.067417 juicefs[7] <WARNING>: AOF is not enabled, you may lose data if Redis is not shutdown properly. [info.go:84]
2024/08/05 20:18:47.067471 juicefs[7] <WARNING>: maxmemory_policy is "volatile-lru",  we will try to reconfigure it to 'noeviction'. [redis.go:3555]
2024/08/05 20:18:47.067767 juicefs[7] <ERROR>: try to reconfigure maxmemory-policy to 'noeviction' failed: ERR unknown command 'config', with args beginning with: 'set' 'maxmemory-policy' 'noeviction'  [redis.go:3557]
2024/08/05 20:18:47.068028 juicefs[7] <INFO>: Ping redis latency: 202.934µs [redis.go:3572]
2024/08/05 20:18:47.069207 juicefs[7] <INFO>: Data use s3://.../juicefs-prod/ [mount.go:605]
2024/08/05 20:18:47.069254 juicefs[7] <WARNING>: cache-size is 0, writeback and prefetch will be disabled [cached_store.go:540]
2024/08/05 20:18:47.072686 juicefs[7] <INFO>: Create session 16887 OK with version: 1.1.0+2023-09-04.08c4ae62 [base.go:492]
2024/08/05 20:18:49.099760 juicefs[7] <INFO>: Prometheus metrics listening on [::]:9567 [mount.go:160]
2024/08/05 20:18:49.100025 juicefs[7] <INFO>: Mounting volume juicefs-prod at /jfs/pvc-fe0548b2-6a6f-4729-8a3e-155018ec5777-tsainc ... [mount_unix.go:269]
2024/08/05 20:18:49.572883 juicefs[7] <INFO>: OK, juicefs-prod is ready at /jfs/pvc-fe0548b2-6a6f-4729-8a3e-155018ec5777-tsainc [mount_unix.go:48]
2024/08/05 20:19:44.053548 juicefs[7] <INFO>: clean up stale session 16880 {Version:1.1.0+2023-09-04.08c4ae62 HostName:juicefs-gke-bc-bc-staging-bc-bc-staging-gpu-l-e49df90a-66f6-pvc IPAddrs:[10.1.1.14] MountPoint:/jfs/pvc-fe0548b2-6a6f-4729-8a3e-155018ec5777-tsainc ProcessID:7}: %!s(<nil>) [base.go:602]
2024/08/05 21:19:10.308785 juicefs[7] <WARNING>: Secret key is removed for the sake of safety [redis.go:3920]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1f4899d]

goroutine 11720 [running]:
github.com/juicedata/juicefs/pkg/object.(*s3client).List(0xc00105f650, {0xc001d68018, 0x12}, {0x0, 0x0}, {0x0, 0x0}, 0x3e8, 0xd0?)
	/workspace/juicefs/pkg/object/s3.go:236 +0x4fd
github.com/juicedata/juicefs/pkg/object.(*withPrefix).List(0xc000b61780, {0x3177db0?, 0xc000091bf8?}, {0x0?, 0x10000c000091c08?}, {0x0, 0x0}, 0x0?, 0x0?)
	/workspace/juicefs/pkg/object/prefix.go:132 +0x112
github.com/juicedata/juicefs/pkg/object.(*withPrefix).List(0xc0016fa040, {0x0?, 0x31df04b?}, {0x0?, 0xc000091db8?}, {0x0, 0x0}, 0x4?, 0x40?)
	/workspace/juicefs/pkg/object/prefix.go:132 +0x112
github.com/juicedata/juicefs/pkg/sync.ListAll({0x3930c60?, 0xc0016fa040}, {0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}, 0x1)
	/workspace/juicefs/pkg/sync/sync.go:108 +0x570
github.com/juicedata/juicefs/pkg/vfs.cleanupBackups({0x392f538?, 0xc000e38f00}, {0xc001dd5fd0?, 0xc0002ba3f0?, 0x55f5d60?})
	/workspace/juicefs/pkg/vfs/backup.go:99 +0xe5
created by github.com/juicedata/juicefs/pkg/vfs.Backup
	/workspace/juicefs/pkg/vfs/backup.go:66 +0x574
....%
```
It seems to break here: https://github.com/juicedata/juicefs/blob/08c4ae6229535e45a73b2a9cc4b80faf01282593/pkg/object/s3.go#L231-L237
**Version**:                                                                                                                                                                                                                                                       
Helm Chart version: 0.19.4 (juicefs-csi-driver)
Juicefs version: v1.1.0

**Comment by zhijian-pro** [comment]
@Haivilo  Did mount add the `--storage-class` parameter?
This may be a bug in v1.1.0, can you test v1.1.3 to see if this is the case?

**Comment by davies** [comment]
@Haivilo This bug is already fixed in v1.1.1, #4183
